server/service/src/purchase_order/update/generate.rs

```diff
@@ -57,38 +65,39 @@ pub fn generate(
         .or(purchase_order.supplier_discount_percentage)
         .unwrap_or(0.0);
 
-    let supplier_discount_amount =
+    updated_order.supplier_discount_percentage = Some(supplier_discount_percentage);
+    updated_order.supplier_discount_amount =
         order_total_before_discount * (supplier_discount_percentage / 100.0);
-    let order_total_after_discount = order_total_before_discount - supplier_discount_amount;
+    updated_order.order_total_after_discount =
+        order_total_before_discount - updated_order.supplier_discount_amount;
 
-    Ok(PurchaseOrderRow {
-        supplier_name_link_id,
-        donor_link_id,
-        reference,
-        status,
-        confirmed_datetime,
-        sent_datetime,
-        contract_signed_date,
-        advance_paid_date,
-        received_at_port_date,
-        requested_delivery_date,
-        supplier_discount_percentage: Some(supplier_discount_percentage),
-        supplier_discount_amount,
-        currency_id,
-        foreign_exchange_rate,
-        shipping_method,
-        comment,
-        order_total_after_discount,
-        ..purchase_order
-    })
+    Ok(updated_order)
 }
 
-fn nullable_update<T: Clone>(
-    input: &Option<crate::NullableUpdate<T>>,
-    current: Option<T>,
-) -> Option<T> {
+fn nullable_update<T: Clone>(input: &Option<NullableUpdate<T>>, current: Option<T>) -> Option<T> {
     match input {
-        Some(crate::NullableUpdate { value }) => value.clone(),
+        Some(NullableUpdate { value }) => value.clone(),
         None => current,
     }
 }
+
+fn set_new_status_datetime(
+    purchase_order: &mut PurchaseOrderRow,
+    input_status: &Option<PurchaseOrderStatus>,
+) {
+    let current_datetime = chrono::Utc::now().naive_utc();
+    if let Some(status) = input_status {
+        match status {
+            PurchaseOrderStatus::Authorised => {
+                println!("add to authorised_datetime to table");
```

Comment:
Note to self: TODO

Reply:
done: https://github.com/msupply-foundation/open-msupply/pull/8690/commits/68c1cec81562d23bd842183b832a7d0401ca020e